great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py

```diff
@@ -0,0 +1,151 @@
+            name=name,
+            data_context=data_context,
+            batch_request=batch_request,
+        )
+
+        self._metric_domain_kwargs = metric_domain_kwargs
+
+        self._threshold = threshold
+        if candidate_strings is not None:
+            self._candidate_strings = set(candidate_strings)
+        else:
+            self._candidate_strings = self.CANDIDATE_STRINGS
+
+        if additional_candidate_strings is not None:
+            self._candidate_strings.update(additional_candidate_strings)
+
+    def _build_parameters(
+        self,
+        parameter_container: ParameterContainer,
+        domain: Domain,
+        *,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+    ):
+        """
+        Check the percentage of values matching each string, and return the best fit, or None if no
+        string exceeds the configured threshold.
+
+        :return: ParameterContainer object that holds ParameterNode objects with attribute name-value pairs and
+        ptional details
+        """
+        validator: Validator = self.get_validator(
+            domain=domain,
+            variables=variables,
+            parameters=parameters,
+        )
+
+        batch_ids: Optional[List[str]] = self.get_batch_ids(
+            domain=domain,
+            variables=variables,
+            parameters=parameters,
+        )
+        if not batch_ids:
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"Utilizing a {self.__class__.__name__} requires a non-empty list of batch identifiers."
+            )
+
+        nonnull_count: int = sum(
+            self.get_metrics(
+                batch_ids=batch_ids,
+                validator=validator,
+                metric_name="column_values.nonnull.count",
+                metric_domain_kwargs=self._metric_domain_kwargs,
+                domain=domain,
+                variables=variables,
+                parameters=parameters,
+            ).metric_values
+        )
+
+        format_string_success_ratios: dict = {}
+        for fmt_string in self._candidate_strings:
+            match_strftime_unexpected_count: int = sum(
+                self.get_metrics(
+                    batch_ids=batch_ids,
+                    validator=validator,
+                    metric_name="column_values.match_strftime_format.unexpected_count",
+                    metric_domain_kwargs=self._metric_domain_kwargs,
+                    metric_value_kwargs={"strftime_format": fmt_string},
+                    domain=domain,
+                    variables=variables,
+                    parameters=parameters,
+                ).metric_values
+            )
+
+            format_string_success_ratios[fmt_string] = (
+                nonnull_count - match_strftime_unexpected_count
+            ) / nonnull_count
+
+        best_fmt_string = None
+        best_ratio = 0
```

Comment:
Could we type annotate as `Optional[str]` and `int`, respectively?

Reply:
✅